src/main/java/net/schmizz/sshj/userauth/password/ConsolePasswordFinder.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C)2009 - SSHJ Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.schmizz.sshj.userauth.password;
+
+import java.io.Console;
+import java.util.IllegalFormatException;
+
+/** A PasswordFinder that reads a password from a console */
+public class ConsolePasswordFinder implements PasswordFinder {
+
+    private final Console console;
+    private final String promptFormat;
+    private final int maxTries;
+
+    private int numTries;
+
+    public static ConsolePasswordFinderBuilder builder() {
+        return new ConsolePasswordFinderBuilder();
+    }
+
+    public ConsolePasswordFinder(Console console, String promptFormat, int maxTries) {
+        this.console = console;
+        this.promptFormat = promptFormat;
```

Comment:
Good catch!  OK, I'll take out the builder (good to have less code to maintain) and put the check here